packages/angular/src/schematics/ngrx/creator-files/__directory__/__fileName__.facade.ts__tmpl__

```diff
@@ -15,7 +14,7 @@ export class <%= className %>Facade {
 
   constructor(private store: Store<from<%= className %>.<%= className %>PartialState>) { }
 
-  loadAll() {
-    this.store.dispatch(<%= className %>Actions.load<%= className %>());
+  dispatch(action: Action) {
+    this.store.dispatch(action);
```

Comment:
Can you expand on this change? My understanding is the benefits of using the facade is that it give you a concrete API of dispatching actions and selecting things. If we simply pass actions through, why having a facade?

Reply:
We still want to encourage good NgRx practices for those who want to use facades.

In general, we recommend:

- Actions should be unique in the source/event they are describing.
- Actions have a unique dispatch point, and are not reused.
- Actions should live close to the place they are dispatched from.

The update to the facade encourages those recommendations.

Reply:
I agree with recommendations. You are saying that the current impl of facades violates "Actions should live close to the place they are dispatched from", right? 

My concern is that as far as I understand facades are usually mocked up in tests like this (I think this is one of the main reasons why folks use facades):

```
it('should create an item', () => {
   const facade = jasmine.createObj();
   const c = new MyComponent(facade);
   c.someClickHander();
   expect(facade.loadBlah).toHaveBeenCalledWith('somevalue');
});
```

with this change you will have to do:

```
it('should create an item', () => {
   const facade = jasmine.createObj();
   const c = new MyComponent(facade);
   c.someClickHander();
   expect(c.dispatch.calls.mostRecent().args[0].valueField).toEqual('somevalue');
});
```
I imagine people who care about facades may not be OK with it.

What do you think?